_posts/2020-03-29-all-you-need-is-lambda-1-booleans.md

```diff
@@ -0,0 +1,260 @@
+---
+layout: post
+title:  "All you need is λ, part one: booleans"
+date:   2020-03-29 20:17:26
+categories: ["lambda calculus", "plt"]
+---
+
+# All you need is λ, part one: booleans
+
+Nearly a hundred years ago, Alonzo Church invented computation in the form of the simple, elegant, and utterly inscrutable lambda calculus. Along with Alan Turing, he then proved that anything computable can be computed in the lambda calculus. However, nearly as soon as we had digital computers, we started inventing programming languages, and with them a vast treasure of features, beautiful and terrible, many of which seem very hard to relate to the fundamental nature of computability, let alone the lambda calculus.
+
+While it’s true that anything which can be computed, period, can be computed in the lambda calculus, you might not want to: it’s austere, to say the least, and was not designed with modern sensibilities regarding readability in mind. We developed all those languages and features for a reason! Still, Church demonstrated not just that it was possible to compute anything computable with the lambda calculus, but also _how_ one might do so.
+
+
+## Unconditional λ
+
+The lambda calculus’ austerity is extreme: you don’t even have booleans. All you have are:
```

Comment:
This is a personal style preference that would feel redundant to some, but I generally prefer numerated/bulleted ideas introduced before their explanations are introduced. This gives my brain a primer of 1) how many things I'm about to read about and 2) what their names are _before_ I learn about them. 

Something like:

> All you have are [Single-argument functions](), [Function application](), [Variables]()

followed by longer definitions/explanations or:


> All you have are:
> 1. [Single-argument functions]()
> 2. [Function application](URL)
> 3. [Variables](url)

and then their longer explanations.